src/base/method/baseManager.ts

```diff
@@ -1,29 +1,26 @@
-import { MethodABI } from '../json';
-import { Method } from './';
-import { MethodContainer } from './container';
+import { BaseMethodContainer } from './baseContainer';
+import { MethodABI } from '../iface';
 
 /**
  * Manages all of the methods for a given contract
  */
-export class MethodManager {
-  private readonly methods: MethodContainerMap;
+export abstract class BaseMethodManager<K extends MethodABI> {
```

Comment:
Similar suggestion as to `BaseMethodContainer`. Please just make this a standalone concrete class and take a `MethodLoader` interface implementation. Nice secondary benefit is that `BaseDecoder` can take a `MethodLoader` as well instead of re-implementing `loadMethods` function